src/window.rs

```diff
@@ -0,0 +1,149 @@
+//! Displays an image in a window created by sdl2.
+
+use image::RgbaImage;
+use sdl2::event::{WindowEvent, Event};
+use sdl2::rect::Rect;
+use sdl2::keyboard::Keycode;
+use sdl2::pixels::{Color, PixelFormatEnum};
+use sdl2::surface::Surface;
+
+/// Displays the provided RGBA image in a new window.
+pub fn display_image(title: &str, image: &RgbaImage, window_width: u32, window_height: u32) {
+
+    let image_width = image.width();
```

Comment:
You could instead write `let (image_width, image_height) = image.dimensions()` if you wanted to save a whole one line.